mk/compile.mk

```diff
@@ -230,4 +230,57 @@ endef
 
 $(foreach f,$(asm-defines-files),$(eval $(call gen-asm-defines-file,$(f),$(out-dir)/$(sm)/include/generated/$(basename $(notdir $(f))).h)))
 
+# Device tree source file compilation
+DTC := dtc
+DTC_FLAGS += -I dts -O dtb
+DTC_FLAGS += -Wno-unit_address_vs_reg
+
+define gen-dtb-file
+# dts file path/name in $1
+# dtb file path/name in $2
+
+dtb-predts-$2 	:= $$(dir $2)$$(notdir $2).pre.dts
+dtb-predep-$2 	:= $$(dir $2)$$(notdir $2).o.d
+dtb-dep-$2	:= $$(dir $2)$$(notdir $2).d
+dtb-cmd-file-$2	:= $$(dir $2).$$(notdir $2).cmd
+
+cleanfiles := $$(cleanfiles) $2 \
+		$$(dtb-predts-$2) $$(dtb-predep-$2) \
+		$$(dtb-dep-$2) $$(dtb-cmd-file-$2)
+
+dtb-cppflags-$2	:= -Icore/include/ -x assembler-with-cpp \
+		   -E -ffreestanding $$(CPPFLAGS) \
+		   -MD -MF $$(dtb-predep-$2) -MT $2
```

Comment:
That should be `-MT $$(dtb-predts-$2)`

Reply:
This current `$2` builds the dependencies for the DTB file.
Changing to `$$(dtb-predts-$2)` breaks the pre-compiled DTS deps on DTB.

Reply:
TL;DR: OK

Well, the `.pre.dts` does not depend on the `.dtb`, it is the opposite. The `.pre.dts` depends on the `.dts` and any included file (`.dtsi` typically) so normally we would want this command to generate: `foo.pre.dts: foo.dts foo_bar.dtsi` etc.
That being said, since we have no separate rule to generate the `.pre.dts` (adding one would make things needlessly more complex), such a dependency would indeed not work properly.